libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1245,6 +1264,36 @@ impl NodeAPI for Greenlight {
         }
     }
 
+    async fn start_keep_alive(&self, mut shutdown: watch::Receiver<()>) {
+        info!("keep alive started");
+        let mut client = self.get_node_client().await.unwrap();
+        let mut interval = tokio::time::interval(Duration::from_secs(15));
+        interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+        loop {
+            tokio::select! {
+              _ = shutdown.changed() => {
+                info!("keep alive exited");
+                break;
+              }
+              _ = interval.tick() => {
+                let inprogress_payments = self.inprogress_payments.load(Ordering::Relaxed);
+                if inprogress_payments == 0 {
+                  continue
```

Comment:
No keep_alive if there are no pending payments?

What if the SDK has no activity for a long time, then the connection is lost, then a payment is attempted?

Reply:
When the payment is attempted (or Any other API) the connection is established, so no issues here.
The only problem is when there is a pending request and then not API call is executed for some time which causes the connection to be killed in the middle of the request

Reply:
Then makes sense.